src/expedited.cpp

```diff
@@ -202,6 +202,14 @@ bool HandleExpeditedBlock(CDataStream &vRecv, CNode *pfrom)
         uint256 blkHash = thinBlock.header.GetHash();
         CInv inv(MSG_BLOCK, blkHash);
 
+        // Message consistency checking
+        if (!IsThinBlockValid(pfrom, thinBlock.vMissingTx, thinBlock.header))
+        {
+            LOCK(cs_main);
+            Misbehaving(pfrom->GetId(), 100);
+            return error("Invalid EXPEDITED_MSG_XTHIN received");
+        }
```

Comment:
I would feel much better if the validity checks for an xthin block and expedited block were the same code.  Expedited is missing much of it.  One is just an unrequested form of the other, so apart from minor details their handling should be identical.